### ISSUE TITLE
Latest version of HPCI puts modules in ${SUDO_USER} rather than my username.

### ISSUE THREAD

**Issue by vanderwb** [NCAR/HPCinstall#81]
```
On 2017-03-13T17:18:33.695340 csgteam (invoked by vanderwb)
called HPCinstall from /glade/u/apps/ch/opt/hpcinstall/1.2/hpcinstall.py
invoked as /glade/u/apps/ch/opt/hpcinstall/1.2/hpcinstall -c build-cdat-lite-6.0.1

Setting environmental variables:
HPCI_SW_DIR       = /glade/u/apps/ch/opt/pythonpkgs/2.7/cdat-lite/6.0.1/intel/16.0.3/
HPCI_SW_NAME      = pythonpkgs/2.7/cdat-lite
HPCI_SW_VERSION   = 6.0.1
HPCI_MOD_DIR      = /glade/u/home/csgteam/CheyenneModules/${SUDO_USER}/CheyenneModules/chey/
HPCI_MOD_DIR_IDEP = /glade/u/home/csgteam/CheyenneModules/${SUDO_USER}/CheyenneModules/chey/idep/
HPCI_MOD_DIR_CDEP = /glade/u/home/csgteam/CheyenneModules/${SUDO_USER}/CheyenneModules/chey/intel/16.0.3/
```

**Comment by davidedelvento** [comment]
@vanderwb on which machine are you running this? I assume Cheyenne given the `ch` in some paths, but asking anyway.
The message `csgteam (invoked by vanderwb)` seems to indicate that this should work since `vanderwb` there is actually `${SUDO_USER}`

**Comment by vanderwb** [comment]
Yes, on Cheyenne.

**Comment by davidedelvento** [comment]
This problem has been introduced in commit b125f8ff8d3bb1f8f74911d190c03ae18925ce33

In that commit the parsing of the config file has been pre-poned to the analysis of "is this a ssh run or not?" logic, becasuse the config file has been added of a parameter needed by that logic. However that logic has the `SUDO_USER` backpropagation in one branch. It's almost catch-22, but not quite, since this branch can be anticipated.

However this is a mess because all of this is happening in the only untested section of the code, which is the one parsing command line arguments. What can go wrong with parsing command line arguments, eh? I should have added tests for that section too, let me see if I can do it quickly while fixing this issue.

**Comment by vanderwb** [comment]
Thanks for the explanation, and for investigating. I will just manually copy the modules to my repo in the meantime.

**Comment by davidedelvento** [comment]
Released as v1.2.1, pushed to github and installed on Cheyenne. The bug is present on Yellowstone and Laramie too, but on the former we don't use `$SUDO_USER` and on the latter is lower priority at the moment, so I did non install it there yet.
